pep-0690.rst

```diff
@@ -328,54 +587,59 @@ startup time and memory usage.
 
 The reference implementation shows small performance regressions in a few
 pyperformance benchmarks, but improvements in others. (TODO update with
-detailed data from 3.11 port of implementation.)
+detailed data from main-branch port of implementation.)
 
 
 How to Teach This
 =================
 
 Since the feature is opt-in, beginners should not encounter it by default.
-Documentation of the ``-L`` flag and ``PYTHONLAZYIMPORTS`` environment variable
-can clarify the behavior of lazy imports.
+Documentation of the ``-L`` flag and ``importlib.enable_lazy_imports()`` can
+clarify the behavior of lazy imports.
+
+The documentation should also clarify that opting into lazy imports is opting
+into an experimental non-standard semantics for Python imports, which could
+cause Python libraries to break in unexpected ways. The responsibility to
+identify these breakages and work around them with an opt-out (or stop using
+lazy imports) rests entirely with the person choosing to enable lazy imports,
+not with the library author. Python libraries are under no obligation to support
+lazy import semantics. Politely reporting an incompatibility may be useful to
+the library author, but they may choose to simply say their library does not
+support use with lazy imports, and this is a valid choice.
 
 Some best practices to deal with some of the issues that could arise and to
 better take advantage of lazy imports are:
 
 * Avoid relying on import side effects. Perhaps the most common reliance on
-  import side effects is the registry pattern, where population of some
-  external registry happens implicitly during the importing of modules, often
-  via decorators. Instead, the registry should be built via an explicit call
-  that perhaps does a discovery process to find decorated functions or classes.
+  import side effects is the registry pattern, where population of some external
+  registry happens implicitly during the importing of modules, often via
+  decorators. Instead, the registry should be built via an explicit call that does
+  a discovery process to find decorated functions or classes in explicitly
+  nominated modules.
 
 * Always import needed submodules explicitly, don't rely on some other import
   to ensure a module has its submodules as attributes. That is, do ``import
-  foo.bar; foo.bar.Baz``, not ``import foo; foo.bar.Baz``. The latter only
-  works (unreliably) because the attribute ``foo.bar`` is added as a side
-  effect of ``foo.bar`` being imported somewhere else. With lazy imports this
-  may not always happen on time.
+  foo.bar; foo.bar.Baz``, not ``import foo; foo.bar.Baz`` (unless there is an
+  explicit import of the submodule in ``foo/__init__.py``.) The latter only works
```

Comment:
The period should be outside of the closing parenthesis, and the parenthetical comment makes the following "latter" reference ambiguous.